Modules/Devices/src/Devices.Infrastructure/Persistence/Database/DevicesDbContext.cs

```diff
@@ -173,5 +173,11 @@ protected override void OnModelCreating(ModelBuilder builder)
         builder.HasDefaultSchema("Devices");
 
         builder.ApplyConfigurationsFromAssembly(typeof(DeviceEntityTypeConfiguration).Assembly);
+
+        builder.Entity<IdentityDeletionProcessAuditLogEntry>()
+            .Property(e => e.MessageKey)
+            .HasConversion(
+                v => v.ToString(),
+                v => (MessageKey)Enum.Parse(typeof(MessageKey), v));
```

Comment:
There is a much easier way of achieving the same goal:

```suggestion
        builder.Entity<IdentityDeletionProcessAuditLogEntry>().Property(e => e.MessageKey).HasConversion<string>();
```

You should try to create a migration to make sure it works. The migration should have no changes.

Reply:
I added the suggested code. Removed the migration, created a new one and it works. Changes made.